gdal/frmts/grib/gribdataset.cpp

```diff
@@ -1125,10 +1125,10 @@ class InventoryWrapperSidecar : public gdal::grib::InventoryWrapper
     explicit InventoryWrapperSidecar(VSILFILE *fp) : gdal::grib::InventoryWrapper()
     {
         VSIFSeekL(fp, 0, SEEK_END);
-        vsi_l_offset length = VSIFTellL(fp);
+        size_t length = static_cast<size_t>(VSIFTellL(fp));
```

Comment:
you shouldn't cast before checking the value. In case a "hostile" .idx file is found